File: ui/app/_components/validation/schema/cruise.ts
import {string, object, number, array} from "yup"
import { yupResolver } from "@hookform/resolvers/yup"

const schema = object().shape({
    startDate: string().required('Start date is required'),
    endDate: string().required('End date is required'),
    routeLength: number(),
    loadsNumber: number(),
    shipId: number().required('Ship must be picked'),
    srcPortId: number().required('Source port is required').typeError('Source port is required'),
    dstPortId: number().required('Destination port is required').typeError('Destination port is required'),
    workersIds: array().required('At least one worker must be on a cruise')
})

const resolver = yupResolver(schema)

export { resolver }

File: api/src/main/java/com/armator/model/Cruise.java
package com.armator.model;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Date;
import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "cruise")
public class Cruise {
    @Id
    @GeneratedValue
    private Integer cruiseId;
    @Temporal(TemporalType.TIMESTAMP)
    private LocalDateTime startDate;
    @Temporal(TemporalType.TIMESTAMP)
    private LocalDateTime endDate;
    private Double routeLength;
    private Integer loadsNumber;
    @ManyToOne
    @JoinColumn(name = "shipId")
    private Ship ship;
    @ManyToOne
    @JoinColumn(name = "srcPortId")
    private Port srcPort;
    @ManyToOne
    @JoinColumn(name = "dstPortId")
    private Port dstPort;
    @ManyToMany(mappedBy = "cruises")
    private Set<Worker> workers;


}
